### PR TITLE
[SPARK-32473][CORE][TESTS] Use === instead IndexSeqView

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to fix `SorterSuite` and `RadixSortSuite` in Scala 2.13 by using `===` instead of `IndexSeqView`.
```
$ git grep "\.view =="
core/src/test/scala/org/apache/spark/util/collection/SorterSuite.scala:    assert(data0.view === data1.view)
core/src/test/scala/org/apache/spark/util/collection/SorterSuite.scala:    assert(data0.view === data2.view)
core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:      assert(ref.view == result.view)
core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:      assert(res1.view == res2.view)
core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:      assert(ref.view == result.view)
core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:      assert(res1.view == res2.view)
```

### Why are the changes needed?

Scala 2.13 reimplements `IndexSeqView` and the behavior is different.
- https://docs.scala-lang.org/overviews/core/collections-migration-213.html

**Scala 2.12**
```scala
Welcome to Scala 2.12.10 (OpenJDK 64-Bit Server VM, Java 1.8.0_262).
Type in expressions for evaluation. Or try :help.

scala> Seq(1,2,3).toArray.view == Seq(1,2,3).toArray.view
res0: Boolean = true
```

**Scala 2.13**
```scala
Welcome to Scala 2.13.3 (OpenJDK 64-Bit Server VM, Java 1.8.0_262).
Type in expressions for evaluation. Or try :help.

scala> Seq(1,2,3).toArray.view == Seq(1,2,3).toArray.view
val res0: Boolean = false
```

### Does this PR introduce _any_ user-facing change?

No. This is a test-only fix.

### How was this patch tested?

- Scala 2.12: Pass the GitHub Action or Jenkins.
- Scala 2.13: Manually test the following.
```
$ dev/change-scala-version.sh 2.13
$ build/mvn test -pl core --am -Pscala-2.13 -Dtest=none -DwildcardSuites=org.apache.spark.util.collection.unsafe.sort.RadixSortSuite
$ build/mvn test -pl core --am -Pscala-2.13 -Dtest=none -DwildcardSuites=org.apache.spark.util.collection.SorterSuite
```

**BEFORE**
```
Tests: succeeded 9, failed 36, canceled 0, ignored 0, pending 0
*** 36 TESTS FAILED ***
Tests: succeeded 3, failed 1, canceled 0, ignored 2, pending 0
*** 1 TEST FAILED ***
```

**AFTER**
```
Tests: succeeded 45, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
Tests: succeeded 4, failed 0, canceled 0, ignored 2, pending 0
All tests passed.
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @srowen and @HyukjinKwon

File: core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala [hunk 1]
```diff
@@ -157,7 +157,7 @@ class RadixSortSuite extends SparkFunSuite with Logging {
         buffer, N, sortType.startByteIdx, sortType.endByteIdx,
         sortType.descending, sortType.signed)
       val result = collectToArray(buffer, outOffset, N)
-      assert(ref.view == result.view)
+      assert(ref.sameElements(result))
```
Comment (HyukjinKwon) [inline core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:160]: Ah, I faced this problem before in spark-xml. I believe you can use `===` instead, see also https://www.scalactic.org/user_guide/DefaultEquality. I remember it reports better when the arrays are not equal. e.g.)

```
scala.Predef.intArrayOps(data0).sameElements[Int](scala.Predef.wrapIntArray(scala.Array.apply(1, 2, 3))) was false
```

vs

```
Expected :Array(1, 2, 3)
Actual   :Array(-2147417843, -2147393996, -2146420562, -2146408050, -2145758434, ...
```

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:160]: Oh, let me try. Thanks

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala:160]: Ur, do you mean to use `IndexSeqView` still like `ref.view === result.view`?

Comment (dongjoon-hyun) [reply core/src/test/scala/org/apache/spark/util/collection/unsafe/sort/RadixSortSuite.scala]: nvm. I check that.

**Comment by dongjoon-hyun** [comment]
The PR is updated according to your review comment, @HyukjinKwon . Thanks always.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon .

**Comment by dongjoon-hyun** [comment]
This is a test-only patch and the changed test suites passed in the GitHub Action core job.
The one failure in `GitHub Action` core job is irrelevant one. Merged to master.

**Review by srowen — COMMENTED** [review]
LGTM

**Comment by dongjoon-hyun** [comment]
Thanks, @srowen .

**Pull request closed without merging**